### ISSUE TITLE
Changing date format

### ISSUE THREAD

**Issue by xunto** [samuelcolvin/pydantic#119]
What is the right way to do it? I used some monkeypatch:
```py
from pydantic import datetime_parse

datetime_parse.date_re = re.compile(r'(?P<day>\d{1,2}).(?P<month>\d{1,2}).(?P<year>\d{4})$')
```
but it feels rather hacky. Looks like I can do it with validators but when I return something It still tries to create date from string after that when it's not a string anymore.

```py
class CheckUserRequest(pydantic.BaseModel):
    cardNumber: str
    lastName: str
    birthday: date

    @validator("birthday", pre=True)
    def parse_birthday(cls, value):
        return datetime.strptime(
            value,
            "%d.%m.%Y"
        ).date()
```

I get:
```py
pydantic.exceptions.ValidationError: error validating input
birthday:
  int() argument must be a string, a bytes-like object or a number, not 'datetime.date' (error_type=TypeError track=date)
```

**Comment by samuelcolvin** [comment]
Not at all sure what you're trying to do. 

Please explain a little more.

**Comment by samuelcolvin** [comment]
Give the input and expected output.

**Comment by xunto** [comment]
@samuelcolvin I use pydantic to validate user request data for my api. I wan't my api to have different format for dates. Like "%d.%m.%Y" instead of "%Y-%m-%d".

**Comment by samuelcolvin** [comment]
Thanks for reporting the solution here is to fix it properly in pydantic, see #122
